AzureAttestationExtension/agent.py

```diff
@@ -0,0 +1,708 @@
+    Note: uninstall operation times out from WAAgent at 5 minutes.
+    We will just try to claen up the attestation client process.
+    Ideally we don't need to do it if disable is called before uninstall.
+    """
+
+    try:        
+        exit_code, output = stop_attestation_client()
+                                            
+    except Exception as ex:
+        exit_code = 1
+        output = 'Uninstall failed with error: {0}\n' \
+                'Stacktrace: {1}'.format(ex, traceback.format_exc())
+
+    return exit_code, output
+
+def enable():
+    """
+    Start the Azure Attestation Linux Agent Service
+    This call will return non-zero or throw an exception if
+    the settings provided are incomplete or incorrect.
+    Note: enable operation times out from WAAgent at 5 minutes
+    """
+       
+    
+
+    public_settings, protected_settings = get_settings()   
+    
+    maa_endpoint = public_settings.get("maaEndpoint")
+    if maa_endpoint is not None:
+        MaaEndpoint = maa_endpoint
+    
+    asc_endpoint = public_settings.get("ascEndpoint")
+    if asc_endpoint is not None:
+        AscReportingEndpoint = asc_endpoint
+
+    # override the config settings if present.    
+    hutil_log_info('Handler initiating onboarding.')
+    try:        
+        exit_code, output = start_attestation_client()
+                                            
+    except Exception as ex:
+        exit_code = 1
+        output = 'Enable failed with error: {0}\n' \
+                'Stacktrace: {1}'.format(ex, traceback.format_exc())
+
+    
+    return exit_code, output
+
+def disable():
+    """
+    Disable Azure Attestation Linux Agent process on the VM.
+    Note: disable operation times out from WAAgent at 15 minutes
+    """
+    
+    try:        
+        exit_code, output = stop_attestation_client()
+                                            
+    except Exception as ex:
+        exit_code = 1
+        output = 'Disable failed with error: {0}\n' \
+                'Stacktrace: {1}'.format(ex, traceback.format_exc())
+
+    return exit_code, output
+
+def update():
+    """
+    Update the current installation of AzureAttestationLinuxAgent
+    No logic to install the agent as agent -> install() will be called 
+    with udpate because upgradeMode = "UpgradeWithInstall" set in HandlerManifest
+    """
+    
+    return 0, ""
+
+# Dictionary of operations strings to methods
+operations = {'Disable' : disable,
+              'Uninstall' : uninstall,
+              'Install' : install,
+              'Enable' : enable,
+              'Update' : update,
+}
+
+def parse_context(operation):
+    """
+    Initialize a HandlerUtil object for this operation.
+    If the required modules have not been imported, this will return None.
+    """
+    hutil = None
+    if ('Utils.WAAgentUtil' in sys.modules
+            and 'Utils.HandlerUtil' in sys.modules):
+        try:
+            
+            logFileName = 'extension.log'
```

Comment:
extension.log is a very generic name. Can't we have something like GuestAttestationExtension.log or AttestationExtension.log?

Reply:
thanks will make the change